drivers/gnss/u_blox_protocol/u_blox_protocol.c

```diff
@@ -0,0 +1,217 @@
+				uint8_t message_class, uint8_t message_id,
+				uint8_t payload[], uint16_t payload_size)
+{
+	uint8_t frame_length_without_payload = 8;
+
+	uint8_t ckA = 0;
+	uint8_t ckB = 0;
+
+	*ubx_frame_size = frame_length_without_payload + payload_size;
+
+	ubx_frame[0] = 0xb5;
+	ubx_frame[1] = 0x62;
+	ubx_frame[2] = message_class;
+	ubx_frame[3] = message_id;
+	ubx_frame[4] = (payload_size & 0xff);
+	ubx_frame[5] = (payload_size >> 8);
+	memcpy(&ubx_frame[6], payload, payload_size);
+
+	for (unsigned int i = 2; i < (*ubx_frame_size - 2); i++) {
+		ckA += ubx_frame[i];
+		ckB += ckA;
+	}
+
+	ubx_frame[*ubx_frame_size - 2] = ckA;
+	ubx_frame[*ubx_frame_size - 1] = ckB;
+}
+
+void u_blox_get_cfg_prt_get(uint8_t ubx_frame[], uint16_t *ubx_frame_size, enum port_number port_id)
+{
+	uint16_t payload_size = 1;
+	uint8_t payload[payload_size];
+
+	payload[0] = port_id;
+
+	u_blox_create_frame(ubx_frame, ubx_frame_size, UBX_CLASS_CFG, UBX_CFG_PRT, payload,
+			    payload_size);
+}
+
+void u_blox_get_cfg_prt_set(uint8_t ubx_frame[], uint16_t *ubx_frame_size, enum port_number port_id,
+			    uint32_t baudrate)
+{
+	uint16_t payload_size = 20;
+	uint8_t payload[payload_size];
+
+	payload[0] = port_id;
+	payload[1] = 0x0;
+
+	payload[2] = 0x0;
+	payload[3] = 0x0;
+
+	/* Mode = 0x8D0 */
+	payload[4] = 0xD0;
+	payload[5] = 0x08;
+	payload[6] = 0x0;
+	payload[7] = 0x0;
+
+	/* Baud Rate */
+	payload[8] = baudrate;
+	payload[9] = baudrate >> 8;
+	payload[10] = baudrate >> 16;
+	payload[11] = baudrate >> 24;
+
+	/* in_proto_mask = All proto enable */
+	payload[12] = 0x7;
+	payload[13] = 0x0;
+
+	/* Out Proto Mask = All proto enable */
+	payload[14] = 0x23;
+	payload[15] = 0x0;
+
+	/* flags */
+	payload[16] = 0x0;
+	payload[17] = 0x0;
+
+	payload[18] = 0x0;
+	payload[19] = 0x0;
+
+	u_blox_create_frame(ubx_frame, ubx_frame_size, UBX_CLASS_CFG, UBX_CFG_PRT, payload,
+			    payload_size);
+}
+
+void u_blox_get_cfg_rst_set(uint8_t ubx_frame[], uint16_t *ubx_frame_size, uint8_t reset_mode)
+{
+	uint16_t payload_size = 4;
+	uint8_t payload[payload_size];
+
+	/* navBbrMask. */
+	payload[0] = 0x00;
+	payload[1] = 0x00;
+
+	/* resetMode. */
+	payload[2] = reset_mode;
+
+	/* reserved1. */
+	payload[3] = 0x00;
+
+	u_blox_create_frame(ubx_frame, ubx_frame_size, UBX_CLASS_CFG, UBX_CFG_RST, payload,
+			    payload_size);
+}
+void u_blox_get_cfg_nav5_get(uint8_t ubx_frame[], uint16_t *ubx_frame_size)
+{
+	u_blox_create_frame(ubx_frame, ubx_frame_size, UBX_CLASS_CFG, UBX_CFG_NAV5, NULL, 0);
+}
+
+void u_blox_get_cfg_nav5_set(uint8_t ubx_frame[], uint16_t *ubx_frame_size, enum gnss_mode g_mode,
+			     enum fix_mode f_mode, int32_t fixed_alt, uint32_t fixed_alt_var,
+			     int8_t min_elev, uint16_t p_dop, uint16_t t_dop, uint16_t p_acc,
+			     uint16_t t_acc, uint8_t static_hold_thresh, uint8_t dgnss_timeout,
+			     uint8_t cno_thresh_num_svs, uint8_t cno_thresh,
+			     uint16_t static_hold_max_dist, enum utc_standard utc_strd)
+{
```

Comment:
This function will push 34 + 36 bytes to the stack on entry, then a few more when entering `u_blox_create_frame`. This can be helped by placing data into the `ubx_frame` buffer directly, and by creating a structure with all the arguments passed to this function, then passing a const pointer to said struct to the function.
```
struct u_blox_protocol_nav_5_cfg {
        enum gnss_mode g_mode;
        enum fix_mode f_mode;
        ...
};

struct u_blox_protocol_nav_5_cfg foo_cfg = {
        .g_mode = ,
        .f_mode = ,
};

ublox_get_nav_5_set(buf, sizeof(buf), &foo_cfg);
```
This also makes it easier to pass the correct arguments to the functions since the names are in the struct, rather than having to count the arguments order (there are 17 arguments, many of which can be mixed up without the compiler noticing, like `p_dop` and `h_dop`)